src/wallet/api/wallet2_api.h

```diff
@@ -220,6 +226,15 @@ struct TransactionInfo
         Direction_Out
     };
 
+    // QUESTION : We already have a bool for "pending" and "failed", but should we rather add two more bools (`m_pending_in_pool` & `m_confirmed`) or switch to an enum like this? I'd prefer the enum, but I guess removing `m_pending` and `m_failed` is not an option?
+    //              If there is justificatin to keep this, any suggestions for a better name? E.g. TxTransmissionState?
```

Comment:
I see this as a typical case for an enum. Easier to process and more logical than 4 booleans that don't tell you at first glance that only exactly one has to be set for things to be consistent.

`TxState` sounds good to me if we don't have something with a stronger "claim" to this name.

How about letting the two existing booleans stand but declaring them as "legacy" and giving the enum as replacement in the comments?